include/Hacl_Hash_SHA3_Scalar.h

```diff
@@ -37,27 +37,52 @@ extern "C" {
 
 void
 Hacl_Hash_SHA3_Scalar_shake128(
-  uint32_t inputByteLen,
-  uint8_t *input,
+  uint8_t *output,
   uint32_t outputByteLen,
-  uint8_t *output
+  uint8_t *input,
+  uint32_t inputByteLen
 );
 
 void
 Hacl_Hash_SHA3_Scalar_shake256(
-  uint32_t inputByteLen,
-  uint8_t *input,
+  uint8_t *output,
   uint32_t outputByteLen,
-  uint8_t *output
+  uint8_t *input,
+  uint32_t inputByteLen
 );
 
-void Hacl_Hash_SHA3_Scalar_sha3_224(uint32_t inputByteLen, uint8_t *input, uint8_t *output);
+void Hacl_Hash_SHA3_Scalar_sha3_224(uint8_t *output, uint8_t *input, uint32_t inputByteLen);
+
+void Hacl_Hash_SHA3_Scalar_sha3_256(uint8_t *output, uint8_t *input, uint32_t inputByteLen);
 
-void Hacl_Hash_SHA3_Scalar_sha3_256(uint32_t inputByteLen, uint8_t *input, uint8_t *output);
+void Hacl_Hash_SHA3_Scalar_sha3_384(uint8_t *output, uint8_t *input, uint32_t inputByteLen);
 
-void Hacl_Hash_SHA3_Scalar_sha3_384(uint32_t inputByteLen, uint8_t *input, uint8_t *output);
+void Hacl_Hash_SHA3_Scalar_sha3_512(uint8_t *output, uint8_t *input, uint32_t inputByteLen);
 
-void Hacl_Hash_SHA3_Scalar_sha3_512(uint32_t inputByteLen, uint8_t *input, uint8_t *output);
+uint64_t *Hacl_Hash_SHA3_Scalar_state_malloc(void);
+
+void Hacl_Hash_SHA3_Scalar_state_free(uint64_t *s);
+
+void
+Hacl_Hash_SHA3_Scalar_shake128_absorb_nblocks(
```

Comment:
These functions need some comments (how to use it) and tests.
But we can do that in a follow-up when it's merged in hacl.

Reply:
I documented the streaming function in Scalar/SIMD256 headers, documents are auto-generated from F* code.